test/server/cards/02_SHD/upgrades/BrutalTraditions.spec.ts

```diff
@@ -0,0 +1,57 @@
+describe('Brutal Traditions', function() {
+    integration(function(contextRef) {
+        describe('Brutal Tradition\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: [
+                            'atst',
+                            'moisture-farmer',
+                            'death-trooper'
+                        ],
+                        discard: ['brutal-traditions']
+                    },
+                    player2: {
+                        groundArena: ['wampa'],
+                        hand: ['confiscate', 'vanquish']
+                    }
+                });
+            });
+
+            it('should be able to play it from the discard pile when an opponent\'s unit is defeated.', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.atst);
+                context.player1.clickCard(context.wampa);
+                context.player2.passAction();
+                expect(context.player1).toBeAbleToSelect(context.brutalTraditions);
```

Comment:
One last check we can do I realized - now that we have your feature for highlighting selectable cards in the action phase, can you add a check that it correctly appears in the highlight list in the action phase? Same for other places we do this check in here